samples/grids/grid/keyboard-navigation-guide/src/index.ts

```diff
@@ -0,0 +1,302 @@
+        <br />
+    </div>`;
+    }
+
+    public changeCombinationsCollection(gridSection: GridSection, evt: any) {
+        switch (gridSection) {
+            case GridSection.THEAD:
+                this.activeCollection = theadKeyCombinations;
+                this.toggleHeaderCombinations(evt);
+
+                break;
+            case GridSection.TBODY:
+                this.activeCollection = tbodyKeyCombinations;
+                this.toggleBodyCombinations(evt);
+                break;
+            case GridSection.FOOTER:
+                this.activeCollection = summaryCombinations;
+                break;
+            default:
+                this.activeCollection = [];
+                return;
+        }
+        this.updateList();
+    }
+
+    public updateList() {
+        this.list.innerHTML = this.listTemplate();
+    }
+
+    public toggleHeaderCombinations(activeNode: any) {
+        const currColumn = this.grid.columns
+            .find(c => c.visibleIndex === activeNode.column && c.level === activeNode.level);
+        this.activeCollection.forEach(x => x.active = true);
+        const actions = this.extractColumnActions(currColumn);
+        this.activeCollection.filter(x => actions.indexOf(x.action) === -1 && x.action !== ItemAction.Always)?.forEach(x => x.active = false);
+    }
+
+    public toggleBodyCombinations(activeNode: any) {
+        const rowRef = this.grid.getRowByIndex(activeNode.row);
+
+        if (rowRef.isGroupByRow) {
+            this.activeCollection.forEach(x => x.active = false);
+            this.activeCollection.filter(x => x.action === ItemAction.Expandable || x.action === ItemAction.Always)?.forEach(x => x.active = true);
+        } else {
+            const currColumn = this.grid.columns.filter(x => !x.columnGroup)
+            .find(c => c.visibleIndex === activeNode.column);
+            const cell = this.grid.getCellByColumn(activeNode.row, currColumn.field);
+            this.toggleCellCombinations(cell);
+        }
+    }
+
+    public toggleCellCombinations(cell?: IgcCellType) {
+        this.activeCollection.forEach(x => x.active = true);
+        const actions = this.extractCellActions(cell);
+        this.activeCollection.filter(x => actions.indexOf(x.action) === -1 && x.action !== ItemAction.Always)?.forEach(x => x.active = false);
+    }
+
+    public extractCellActions(cell: IgcCellType) {
+        const res: any[] = [];
+        if(!cell) return res;
+        if (cell?.editable) {
+            res.push(ItemAction.Editable);
+        }
+
+        res.push(ItemAction.Collapsible);
+        return res;
+    }
+
+    public extractColumnActions(col: IgcColumnComponent | IgcColumnGroupComponent) {
+        const res = [];
+        if (col.sortable) {
+            res.push(ItemAction.Sortable);
+        }
+
+        if (col.filterable && !col.columnGroup) {
+            res.push(ItemAction.Filterable);
+        }
+
+        if (col.collapsible) {
```

Comment:
Only this needs the col to be casted to `IgcColumnGroupComponent`, otherwise throws error.